### PR TITLE
:shield: pos_scan_ref updated tests. cover the basic functionality

### PR DESCRIPTION
**Pull request by Ommo73**

### CODE REVIEW

**Review by KolushovAlexandr — CHANGES_REQUESTED** [review]
check and fix incompatibility with pos_cashier_select in the same pr with tests

File: pos_scan_ref/static/src/js/test_pos_scan_ref.js [hunk 1]
```diff
@@ -11,45 +11,27 @@ odoo.define('pos_scan_ref.tour', function (require) {
...
             trigger: '.order-button.neworder-button',
+        }, {
+            content: 'Scan product by reference code',
+            trigger: '.pos-branding',
+            run: function () {
+                posmodel.barcode_reader.scan(refcode);
+            },
+        }, {
+            content: 'Check that the product was scanned successfully',
+            trigger: '.product-name:contains("Boni Oranges")',
```
Comment (KolushovAlexandr) [inline pos_scan_ref/static/src/js/test_pos_scan_ref.js]: It doesn't guarantee that the orderline was added

File: pos_scan_ref/static/src/js/test_pos_scan_ref.js [hunk 2]
```diff
@@ -14,11 +14,11 @@ odoo.define('pos_scan_ref.tour', function (require) {
         }, {
             trigger: ".o_pos_kanban button.oe_kanban_action_button",
             content: _t("<p>Click to start the point of sale interface. It <b>runs on tablets</b>, laptops, or industrial hardware.</p><p>Once the session launched, the system continues to run without an internet connection.</p>"),
-            position: "bottom",
+            position: "bottom"
```
Comment (KolushovAlexandr) [inline pos_scan_ref/static/src/js/test_pos_scan_ref.js]: these changes are not related to the commit message

File: pos_product_lot/static/src/js/pos.js [hunk 3]
```diff
@@ -93,16 +94,18 @@ function geByClass(searchClass, node, tag) {
...
 
-            if (parsed_code.type === 'price') {
-                selectedOrder.addProduct(product, {price:parsed_code.value});
-            } else if (parsed_code.type === 'weight'){
-                selectedOrder.addProduct(product, {quantity:parsed_code.value, merge:false});
-            } else if (parsed_code.type === 'discount'){
-                selectedOrder.addProduct(product, {discount:parsed_code.value, merge:false});
-            } else {
-                selectedOrder.addProduct(product);
+            if(parsed_code.type === 'price'){
```
Comment (KolushovAlexandr) [inline pos_product_lot/static/src/js/pos.js]: Please get rid of dispensable changes

File: pos_product_lot/static/src/js/pos.js [hunk 4]
```diff
@@ -93,16 +94,18 @@ function geByClass(searchClass, node, tag) {
...
+                selectedOrder.add_product(product, {price:parsed_code.value});
```
Comment (KolushovAlexandr) [inline pos_product_lot/static/src/js/pos.js:102]: add it to the changelog as a fix

**Review by KolushovAlexandr — CHANGES_REQUESTED** [review]

File: pos_product_lot/static/src/js/pos.js (hunk 4 above)
Comment (Ommo73) [reply pos_product_lot/static/src/js/pos.js:102]: DONE

File: pos_product_lot/static/src/js/pos.js (hunk 3 above)
Comment (KolushovAlexandr) [reply pos_product_lot/static/src/js/pos.js]: I meant not only this one, please look at the ``Files Changed`` and fix all of them

**Review by KolushovAlexandr — APPROVED** [review]

**Comment by KolushovAlexandr** [comment]
@Ommo73 Who's next to check it?

**Comment by yelizariev** [comment]
Чей ход?

**Comment by em230418** [comment]
Maybe :cherries: for https://github.com/it-projects-llc/pos-addons/pull/976/commits/bb8db5f7796cb07462d3a3b5ae9e0e85ab04f319 ? And  :ambulance: for https://github.com/it-projects-llc/pos-addons/pull/976/commits/141e753d1d50523a66717ba5ed74b8184b8b6989 ?

**Comment by em230418** [comment]
```
Changelog: `<doc/changelog.rst>`_
Use template: "Tested on Odoo {VERSION} {ODOO_COMMIT_SHA_TO_BE_UPDATED}"
************* Module pos_product_lot
pos_product_lot/__init__.py:1: [E7989(readme-versions), ] Wrong version in module's README.rst
pos_product_lot/__init__.py:1: [W7993(js-empty-coverage), ] JS files are not covered (phantom_js is not used). Please add js tests: https://odoo-development.readthedocs.io/en/latest/qa/js/index.html
```

**Comment by KolushovAlexandr** [comment]
@em230418 about this 'And  ambulance for 141e753 ?' It worked as it should, the only reason it was changed - failed tests

**Comment by KolushovAlexandr** [comment]
@em230418 also, I don't think that this is necessary 
```
pos_product_lot/__init__.py:1: [W7993(js-empty-coverage), ] JS files are not covered (phantom_js is not used). Please add js tests: https://odoo-development.readthedocs.io/en/latest/qa/js/index.html
```

**Comment by em230418** [comment]
> @em230418 about this 'And ambulance for 141e753 ?' It worked as it should, the only reason it was changed - failed tests

Ok. Accepted.

> @em230418 also, I don't think that this is necessary
If travis passes without it - that is fine for me.

**Review by em230418 — APPROVED** [review]

**Comment by em230418** [comment]
I approve to merge it now

**Pull request merged**